lib/at_ex/gateway/Payments/Mobile/b2c.ex

```diff
@@ -0,0 +1,104 @@
+defmodule AtEx.Gateway.Payments.Mobile.B2c do
+  @moduledoc false
+
+  # Using this system for delivery of which URL to use (sandbox or live)
+  # determined by whether we are in production or development or test
+  # Selection of the live URL can be forced by setting an environment
+  # variable FORCE_AT_LIVE=YES
+  url =
+    if Mix.env() == :prod || System.get_env("FORCE_AT_LIVE") == "YES" do
+      "https://payments.africastalking.com/mobile"
+    else
+      "https://payments.sandbox.africastalking.com/mobile"
+    end
+
+  # The `type` config is to allow the api send `application/json` check https://github.com/teamon/tesla#formats for more info
+
+  use AtEx.Gateway.Base, url: url, type: "json"
+
+  @doc """
+  This function initiates a mobile B2C request by making a HTTP POST request to the Africa's talking Mobile B2C endpoint.
+
+  ## Parameters
+  attrs: - a list of Recipient each a map containing a `phoneNumber`, `currencyCode`, `amount` and a map of `metadata` key optionally it may also contain `name`, `reason` and `providerChannel` see the docs at https://build.at-labs.io/docs/payments%2Fmobile%2Fb2c%2Foverview for how to use these keys
+
+  ## Example 
+    iex>AtEx.Gateway.Payments.Mobile.B2c.b2c_checkout([%{phoneNumber: "254724540000", amount: 10, currencyCode: "KES", metadata: %{ message: "I am here"}}])
+    {:ok,
+  %{
+   "entries" => [
+     %{
+       "phoneNumber" => "+254724540000",
+       "provider" => "Athena",
+       "providerChannel" => "525900",
+       "status" => "Queued",
+       "transactionFee" => "KES 0.1000",
+       "transactionId" => "ATPid_2b76da39eebd5c6bcc49e5d30c3d0390",
+       "value" => "KES 10.0000"
+     }
+   ],
+   "numQueued" => 1,
+   "totalTransactionFee" => "KES 0.1000",
+   "totalValue" => "KES 10.0000"
+  }}
+  """
+  @spec b2c_checkout(list()) :: {:ok, term()} | {:error, term()}
+  def b2c_checkout(attrs) when is_list(attrs) and length(attrs) <= 10 do
+    username = Application.get_env(:at_ex, :username)
+    product_name = Application.get_env(:at_ex, :b2c_product_name)
+
+    case validate_attrs(attrs) do
+      %{ok: _} ->
+        params =
+          %{recipients: attrs}
+          |> Map.put(:username, username)
+          |> Map.put(:productName, product_name)
+
+        "/b2c/request"
+        |> post(params)
+        |> process_result()
+
+      %{error: message} ->
```

Comment:
With `process_result()` I think we already cover the error processing, just confirm though.

Reply:
this is validation response, for required params, the api call response is processed by  `process_result()`